packages/host/app/components/card-catalog/index.gts

```diff
@@ -1,103 +1,119 @@
-                    @title={{card.title}}
-                    @description={{card.description}}
-                    @thumbnailURL={{card.thumbnailURL}}
-                  />
-                  <button
-                    class='select'
-                    {{on 'click' (fn @select card)}}
-                    {{on 'dblclick' (fn @select card)}}
-                    {{on 'keydown' (fn this.handleEnterKey card)}}
-                    data-test-select={{card.id}}
-                    aria-label='Select'
-                  />
-                  <IconButton
-                    class='hover-button preview'
-                    @icon={{EyeIcon}}
-                    aria-label='preview'
-                  />
-                </li>
-              {{/let}}
-            {{/each}}
-          </ul>
-          {{#if (gt realm.cards.length realm.displayedCards.length)}}
-            <Button
-              {{on 'click' (fn this.displayMoreCards realm)}}
-              @kind='secondary-light'
-              @size='small'
-              data-test-show-more-cards
-            >
-              Show more cards
-            </Button>
-          {{/if}}
-        </section>
-      {{else}}
-        <p>No cards available</p>
-      {{/each}}
+
+      {{#let
+        (this.groupByRealmUrl @cards this.pageSize)
+        as |groupedCardsByRealm|
+      }}
+        {{#each-in groupedCardsByRealm as |_realmUrl realmData|}}
+          <section
+            class='card-catalog__realm'
+            data-test-realm={{realmData.realmInfo.name}}
+          >
+            <CardCatalogResultsHeader
+              @realm={{realmData.realmInfo}}
+              @resultsCount={{realmData.cardsTotal}}
+            />
+
+            <ul class='card-catalog__group'>
+              {{#each realmData.cards as |card index|}}
+                {{#if (lt index realmData.displayedCardsCount)}}
+                  {{#let (eq @selectedCardUrl card.url) as |isSelected|}}
+                    <li class={{cn 'item' selected=isSelected}}>
+                      <button
+                        class='catalog-item {{if isSelected "selected"}}'
+                        {{on 'click' (fn @select card.url)}}
+                        {{on 'dblclick' (fn @select card.url)}}
+                        {{on 'keydown' (fn this.handleEnterKey card.url)}}
+                        data-test-select={{removeFileExtension card.url}}
+                        aria-label='Select'
+                        data-test-card-catalog-item={{removeFileExtension
+                          card.url
+                        }}
+                        data-test-card-catalog-item-selected={{isSelected}}
+                      >
+                        {{card.component}}
+                      </button>
+                    </li>
+                  {{/let}}
+                {{/if}}
+              {{/each}}
+            </ul>
+
+            {{#if (gt realmData.cardsTotal realmData.displayedCardsCount)}}
+              <Button
```

Comment:
I added some helpful info. Not in the design but I think it's quite useful to get a sense of quantities:

Before:
<img width="184" alt="image" src="https://github.com/user-attachments/assets/9706160c-d233-4dd6-9f72-9091ee48697c">

After:
<img width="319" alt="image" src="https://github.com/user-attachments/assets/c4a7b2ba-67a6-4af5-8f9e-26e57ddf22e9">